apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/Playground/index.tsx

```diff
@@ -1,47 +1,41 @@
 'use client'
 
-import {
-  Dispatch,
-  SetStateAction,
-  useCallback,
-  useEffect,
-  useMemo,
-  useState,
-} from 'react'
+import { useCallback, useEffect, useState } from 'react'
 
 import { ConversationMetadata } from '@latitude-data/compiler'
 import { DocumentVersion } from '@latitude-data/core/browser'
-import { Badge, Input, Text } from '@latitude-data/web-ui'
+import { Badge, Input, Text, useCurrentCommit } from '@latitude-data/web-ui'
+import {
+  PlaygroundInputs,
+  useDocumentParameters,
+} from '$/hooks/useDocumentParameters'
 
 import { Header } from '../Header'
 import Chat from './Chat'
 import Preview from './Preview'
 
-export function convertParams(inputs: Record<string, string>) {
-  return Object.fromEntries(
-    Object.entries(inputs).map(([key, value]) => {
-      try {
-        return [key, JSON.parse(value)]
-      } catch (e) {
-        return [key, value]
-      }
-    }),
-  )
-}
-
 function InputParams({
   inputs,
   setInputs,
 }: {
-  inputs: Record<string, string>
-  setInputs: Dispatch<SetStateAction<Record<string, string>>>
+  inputs: PlaygroundInputs
+  setInputs: (inputs: PlaygroundInputs) => void
 }) {
+  const [ssr, setSsr] = useState(true)
   const setInput = useCallback(
     (param: string, value: string) => {
       setInputs({ ...inputs, [param]: value })
     },
-    [inputs],
+    [inputs, setInputs],
   )
+  useEffect(() => {
+    setSsr(false)
+  }, [])
+
+  // Avoid rendering inputs on server
+  // We have a Hydration issue with the inputs becasue
```

Comment:
typo becasue

Reply:
but i don't get it this has the 'use client' pragma

Reply:
Client components are also rendered on the server. Client pragma means react will hydrate this component